workspaces/vscode-cpp/Dockerfile

```
@@ -0,0 +1,31 @@
+ARG BASE_IMAGE=prairielearn/workspace-vscode-base
+ARG BASE_IMAGE_TAG=latest
+
+FROM ${BASE_IMAGE}:${BASE_IMAGE_TAG}
+
+USER root
+RUN apt-get update && apt-get upgrade -y \
+    && apt-get install -y --no-install-recommends \
+    build-essential \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && find /tmp -not -path /tmp -delete
+
+USER coder
+
+# After installing C/C++ we install some VS Code extensions.
+# codelldb is installed from a downloaded file because the open-vsx version of
+# the extension downloads the latest version of the extension upon loading,
+# which would not be appropriate in a containerized environment with restricted
+# network access.
+RUN arch=$(uname -m) && \
+    if [ "$arch" = "x86_64" ]; then arch="x64"; \
+    elif [ "$arch" = "aarch64" ]; then arch="arm64"; \
+    else echo "Unsupported architecture: $arch"; exit 1; fi && \
```

To double-check what the extension itself uses to identify the platform I checked this:
https://github.com/vadimcn/codelldb/blob/d1ae2ebef7e44ea13d3aa8282f0b0b85d4838b6f/extension/install.ts#L96-L103
https://github.com/vadimcn/codelldb/blob/d1ae2ebef7e44ea13d3aa8282f0b0b85d4838b6f/package.json#L1124-L1135

Since we're working with a guaranteed Linux environment and we're assuming that the build only happens in either amd64 or arm64 I think this approach makes sense, however an alternative option may be to use:

```suggestion
RUN arch=$(/usr/lib/code-server/lib/node -p 'os.arch()') && \
```

Won't block on this, though.

---

I would prefer to stick with `uname -m` for consistency reasons.